lib/snowflex/db_connection.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Snowflex.DBConnection do
+  @moduledoc """
+  Defines a Snowflake connection with DBConnection adapter.
+
+  ## Definition
+
+  When used, the connection expects the `:otp_app` option. You may also define a standard timeout.
+  This will default to 60 seconds.
+
+  ```
+  defmodule SnowflakeDBConnection do
+    use Snowflex.DBConnection,
+      otp_app: :my_app,
+      timeout: :timer.seconds(60)
+  end
+  ```
+  """
+
+  alias Snowflex.DBConnection.{
+    Protocol,
+    Query
+  }
+
+  @doc false
+  defmacro __using__(opts) do
+    quote bind_quoted: [opts: opts] do
+      # setup compile time config
+      otp_app = Keyword.fetch!(opts, :otp_app)
+      timeout = Keyword.get(opts, :timeout, :timer.seconds(60))
+
+      @otp_app otp_app
+      @timeout timeout
+      @name __MODULE__
+
+      def child_spec(_) do
```

Comment:
why do we ignore child spec input options?

Reply:
`opts` should be provided statically when configuring your connection module with `use Snowflake.DBConnection, opt1: _, opt2: _, ...`. Ignoring the opts in `child_spec/1` ensures that the static config in the `use` macro is always correct, and not accidentally overridden in the supervision tree. I like this approach 👍

Reply:
gotcha, thanks